olp-cpp-sdk-authentication/include/olp/authentication/CredentialsHelper.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+
+#include "olp/authentication/TokenProvider.h"
+
+namespace olp {
+namespace authentication {
+
+/*
+ * @brief Reads credentials to access OLP from an input stream that is
+ * interpreted as a sequence of characters.
+ *
+ * The stream must contain the following sequences of characters:
+ * here.access.key.id = <user's key id>
+ * here.access.key.secret = <user's key secret>
+ *
+ * @param[in] stream Stream to read the credentials from.
+ * @param[in] network_request_handler The network instance to be used to
+ * initialize token provider.
+ * @param[in] stream Stream to read the credentials from.
+ * @return TokenProvider that uses credentials from the file. The caller should
+ * check the returned token provider for validity. It may be invalid in case of
+ * broken stream or invalid credentials.
+ */
+TokenProviderDefault ReadFromStream(
+    const std::shared_ptr<http::Network>& network_request_handler,
```

Comment:
Passing network to this function seems strange. You may create factory function, that creates `AuthenticationCredentials` from filepath or stream. With interfaces something like:
```cpp
client::ApiResponse<AuthenticationCredentials, client::ApiError> ParseCredentials(std::istream& stream);
client::ApiResponse<AuthenticationCredentials, client::ApiError> ParseCredentials(const std::string& filename);
```
ApiResponse is used as container for result or error.

Reply:
Changed return type to `boost::optional<AuthenticationCredentials>` because `client::ApiResponse` doesn't accept types that has no default constructor.